src/System.Private.CoreLib/src/System/DateTime.Windows.cs

```diff
@@ -0,0 +1,175 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System
+{
+    public readonly partial struct DateTime
+    {
+        internal static readonly bool s_systemSupportsLeapSeconds = SystemSupportsLeapSeconds();
+
+        public static DateTime UtcNow
+        {
+            get
+            {
+                if (s_systemSupportsLeapSeconds)
+                {
+                    FullSystemTime time;
+                    GetSystemTimeWithLeapSecondsHandling(out time);
+                    return CreateDateTimeFromSystemTime(in time);
+                }
+
+                return new DateTime(((ulong)(GetSystemTimeAsFileTime() + FileTimeOffset)) | KindUtc);
+            }
+        }
+
+        public static DateTime FromFileTimeUtc(long fileTime)
```

Comment:
The only difference between Windows and Unix versions of FromFileTimeUtc and ToFileTimeUtc is the callout to the special method method for leap seconds. Should we have keep it in the shared and have a dummy method on Unix?

Also, it may be better to call InternalFromFileTime / InternalToFileTime something like `FromFileTimeLeapSecondsAware / ToFileTimeLeapSecondAware`.